procedures/unit-testing-test-md-gen.ipf

```diff
@@ -245,3 +270,71 @@ static Function/WAVE CheckDGenOutput(procWin, fullFuncName, dgen, wType0, wType1
 
 	return wGenerator
 End
+
+static Function ExecuteAllDataGenerators()
+	variable i, j, size, length, dimPos, succeed
+	string dgenList, dgen, fullFuncName, endTime, msg
+	variable DGENLIST_Index
+
+	WAVE/T testRunData = UTF_Basics#GetTestRunData()
+	size = DimSize(testRunData, UTF_ROW)
+	DGENLIST_Index = FindDimLabel(testRunData, UTF_COLUMN, "DGENLIST")
+
+	if(!size)
+		return NaN
+	endif
+
+	WAVE/WAVE wDgen = GetDataGeneratorWaves()
+
+	UTF_Reporting_Control#TestSuiteBegin("@DGEN_SUITE")
+
+	for(i = 0; i < size; i += 1)
+		dgenList = testRunData[i][DGENLIST_Index]
+		WAVE/T dgenWave = ListToTextWave(dgenList, ";")
+		length = DimSize(dgenWave, UTF_ROW)
+
+		for(j = 0; j < length; j += 1)
+			dgen = dgenWave[j]
+			dimPos = FindDimLabel(wDgen, UTF_ROW, dgen)
+			if(dimPos >= -1)
+				continue
+			endif
+
+			FUNCREF TEST_CASE_PROTO_DGEN fDgen = $dgen
+			if(!UTF_FuncRefIsAssigned(FuncRefInfo(fDgen)))
+				fullFuncName = testRunData[i][%FULLFUNCNAME]
+				sprintf msg, "Data Generator function %s has wrong format. It is referenced by test case %s.", dgen, fullFuncName
+				UTF_Reporting#ReportErrorAndAbort(msg)
+			endif
+
+			succeed = 1
```

Comment:
I think you don't need a success state variable here, that is also initialized to success.
The information you want to track is if you add the data generator or not to the wave.